tests/test_typecheck.py

```diff
@@ -80,14 +80,14 @@ def foo(n: size, A: R[n] @ GEMM_SCRATCH):
 
 def test_sin1():
     @proc
-    def sin(x: f32):
+    def sin_proc(x: f32):
         y: f32
         y = sin(x)
 
 
 def test_sin2():
     @proc
-    def sin(x: f32):
+    def sin_proc(x: f32):
```

Comment:
Was this change necessary? Is it because the procedure name is conflicting with the extern name? 
I think procedure definitions should be able to overwrite whatever other global and local variables out there

Reply:
The change is necessary because the sin function call inside the proc is parsed as an implicit unquote in the metalanguage, which means it is looked up like a normal python variable named sin would be looked up at that point. But, if the proc is named sin, it is registered as a local variable in the scope which shadows the sin extern that is imported as a global variable.

Reply:
Hmmmm I see what you mean but this seems quite cryptic.... It's way too opaque to users... How bad is it going to be if we don't support implicit unquotes?

Reply:
I think the code becomes pretty cluttered if there are no implicit unquotes because you will have to put braces around every variable you reference from the metalanguage.

Also, I think it should be expected that if you name a proc "sin" it will shadow any other "sin" inside its scope. In most other languages, you can refer to the function you are currently inside by its name (e.g. recursion), so it doesn't seem very surprising.

Reply:
As a side note, if you try compiling this exo:
```
@proc
def sin(x: f32):
    x = sin(x)
```
the result is:
```
#include <math.h>
// sin(
//     x : f32 @DRAM
// )
void sin( void *ctxt, float* x ) {
*x = sin((float)*x);
}
```
which is actually invalid C because the `sin` inside the function will refer to the proc `sin` not the function from `math.h`.
```

Reply:
I see okay, this makes sense. Can you add a test case of declaring a proc named `sin` that shadows the extern sin, and call the proc `sin` from another procedure `foo`, and see if that works as expected?

Reply:
complete